core/src/org/pentaho/di/core/logging/LoggingBuffer.java

```diff
@@ -260,6 +271,31 @@ public void removeChannelFromBuffer( String id ) {
     }
   }
 
+  public void waitSpoonLogTabRefresherBeforeRemoving() {
+    // check environment
+    if ( KettleClientEnvironment.ClientType.SPOON.equals( KettleClientEnvironment.getInstance().getClient() ) ) {
+      // check threads
+      String currentTreadName = Thread.currentThread().getName();
+      if ( Const.JOB_EXECUTOR_DISCARD_LINES_THREAD.equals( currentTreadName )
+        || Const.TRANS_EXECUTOR_DISCARD_LINES_THREAD.equals( currentTreadName ) ) {
+        // set timeout expiration for discard lines waiting
+        long timeoutExpired = System.currentTimeMillis() + Const.KETTLE_DISCARD_LOG_LINES_WAIT_TIMEOUT;
+        // wait until log tab refresher reads a new logs
+        while ( !isBufferReadBySpoonLogTabRefresher ) {
+          try {
+            buffer.wait( timeoutExpired - System.currentTimeMillis() );
+          } catch ( InterruptedException e ) {
+            e.printStackTrace();
```

Comment:
Definitely don't want this line at all. Perhaps if Debug logging was enabled, but even then it's unlikely. This seems like a debug message to me.